### PR TITLE
Update CACHE_VERSION of primer, add comment and update paths

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Similar to #5602.

Also updates the primer to run when changes to its workflow file occur.

Should fix CI issues of #5609 (already tested and it works).

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
   pull_request: ~
 
 env:
-  CACHE_VERSION: 4
+  CACHE_VERSION: 4 # Also change this in the primer workflow
```
Comment (cdce8p) [inline .github/workflows/ci.yaml:11]:
```suggestion
  CACHE_VERSION: 4  # Also change this in the primer workflow
```
It's usually common to put 2 whitespaces between code and a comment, same as in Python.

File: .github/workflows/primer-test.yaml [hunk 2]
```diff
@@ -10,9 +10,10 @@ on:
       - "pylint/**"
       - "tests/primer/**"
       - "requirements*"
+      - ".github/workflows/primer-test.yaml"
 
 env:
-  CACHE_VERSION: 3
+  CACHE_VERSION: 4 # Also change this in the CI workflow
```
Comment (cdce8p) [inline .github/workflows/primer-test.yaml:16]:
```suggestion
  CACHE_VERSION:  4 # Also change this in the CI workflow
```

**Review by cdce8p — APPROVED** [review]
Just a small comment

**Review by Pierre-Sassoulas — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -8,7 +8,7 @@ on:
   pull_request: ~
 
 env:
-  CACHE_VERSION: 4
+  CACHE_VERSION: 4  # Also change this in the primer workflow
```
Comment (cdce8p) [inline .github/workflows/ci.yaml:11]:
```suggestion
  # Also change CACHE_VERSION in the primer workflow
  CACHE_VERSION: 4
```
`prettier` doesn't seem to like two spaces 😞

File: .github/workflows/primer-test.yaml [hunk 4]
```diff
@@ -10,9 +10,10 @@ on:
       - "pylint/**"
       - "tests/primer/**"
       - "requirements*"
+      - ".github/workflows/primer-test.yaml"
 
 env:
-  CACHE_VERSION: 3
+  CACHE_VERSION: 4  # Also change this in the CI workflow
```
Comment (cdce8p) [inline .github/workflows/primer-test.yaml:16]:
```suggestion
  # Also change CACHE_VERSION in the CI workflow
  CACHE_VERSION: 4
```

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (DanielNoord) [reply .github/workflows/ci.yaml:11]: Oh wel.. 😢

**Pull request merged**